src/sentry/taskworker/service/service.py

```diff
@@ -18,47 +20,18 @@ class TaskService:
     can rely on this interface to be stable.
     """
 
-    def get_task(self, *, partition: int | None = None, topic: str | None = None) -> RpcTask | None:
-        logger.info("getting_latest_tasks", extra={"partition": partition, "topic": topic})
-        with transaction.atomic(using=router.db_for_write(PendingTasks)):
-            query_set = PendingTasks.objects.filter(state=PendingTasks.States.PENDING)
-
-            if partition is not None:
-                query_set = query_set.filter(partition=partition)
-
-            if topic is not None:
-                query_set = query_set.filter(topic=topic)
-
-            task = query_set.first()
-            if task is None:
-                return None
-
-            # TODO this duration should be a tasknamespace setting, or with an option
-            deadline = task.added_at + timedelta(minutes=3)
+    def __init__(self):
+        self.pending_task_store = PendingTaskStore()
 
-            task.update(state=PendingTasks.States.PROCESSING, processing_deadline=deadline)
-            return serialize_task(task)
-
-    def set_task_status(self, *, task_id: int, task_status: PendingTasks.States) -> RpcTask | None:
-        try:
-            with transaction.atomic(using=router.db_for_write(PendingTasks)):
-                # Pull a select for update here to lock the row while we mutate the retry count
-                task = PendingTasks.objects.select_for_update().filter(id=task_id).get()
-
-                # TODO add state machine validation/logging
-                if task.state == PendingTasks.States.COMPLETE:
-                    return serialize_task(task)
-
-                task.update(state=task_status)
-                if task_status == PendingTasks.States.RETRY:
-                    task.update(retry_attempts=task.retry_attempts + 1)
+    def get_task(self, *, partition: int | None = None, topic: str | None = None) -> Task | None:
+        logger.info("getting_latest_tasks", extra={"partition": partition, "topic": topic})
+        return self.pending_task_store.get_pending_task(partition, topic)
 
-            return serialize_task(task)
-        except PendingTasks.DoesNotExist:
-            return None
+    def set_task_status(self, *, task_id: int, task_status: Status.ValueType):
+        self.pending_task_store.set_task_status(task_id=task_id, task_status=task_status)
 
-    def complete_task(self, *, task_id: int) -> RpcTask | None:
-        return self.set_task_status(task_id=task_id, task_status=PendingTasks.States.COMPLETE)
+    def complete_task(self, *, task_id: int):
+        self.set_task_status(task_id=task_id, task_status=COMPLETE)
```

Comment:
Removed the return value here because I'm not clear what we're trying to accomplish with the return value (nothing uses it)